src/Nethermind/Nethermind.HealthChecks/FreeDiskSpaceChecker.cs

```diff
@@ -80,5 +65,44 @@ public async ValueTask DisposeAsync()
             await StopAsync(default);
             _timer.Dispose();
         }
+
+        public void EnsureEnoughFreeSpaceOnStart(ITimerFactory timerFactory)
+        {
+            float minAvailableSpaceThreshold = 2 * _healthChecksConfig.LowStorageSpaceShutdownThreshold;
+            if (_healthChecksConfig.LowStorageSpaceWarningThreshold > 0 && _healthChecksConfig.LowStorageSpaceWarningThreshold > _healthChecksConfig.LowStorageSpaceShutdownThreshold)
+                minAvailableSpaceThreshold = _healthChecksConfig.LowStorageSpaceWarningThreshold / 2.0f;
+
+            if (!IsEnoughDiskSpace(minAvailableSpaceThreshold))
+            {
+                ManualResetEventSlim mre = new(false);
+                using ITimer timer = timerFactory.CreateTimer(TimeSpan.FromMinutes(CheckPeriodMinutes));
```

Comment:
Not sure if FilleSystemWatcher could be used instead of timer, with proper NotifyFilters and root directory of the disk. Might not work.

Reply:
I was looking into that, but don't think this would be more cumbersome, would have to track all deletions from root.